### PR TITLE
add warning modal for debug wallets

### PR DESCRIPTION
**Pull request by v-almonacid**
Modal is added right after a wallet is selected, within the TxHistory screen. Accepting the modal sends user to WalletSelectionScreen.

Edit: modal is also added in Delegation dashboard. In both cases it's triggered if wallet corresponds to any of the two default wallets used for debugging.

### CODE REVIEW

**Review by nicarq — APPROVED** [review]
🙌

File: src/crypto/wallet.js [hunk 1]
```diff
@@ -1172,6 +1172,36 @@ class WalletManager {
...
+
+    for (const mnemonic of mnemonics) {
+      Logger.debug('WalletManager::checkForFlawedWallets mnemonic:', mnemonic)
+      const flawedAddresses = await util.getAddressesFromMnemonics(
+        mnemonic,
+        'External',
+        [0, 1, 2, 3, 4],
+        CARDANO_CONFIG.MAINNET,
+      )
+      for (let i = 0; i < flawedAddresses.length; i++) {
```
Comment (SebastienGllmt) [inline src/crypto/wallet.js]: is this not a little excessive? Can't you just do
```javascript
    if (this._wallet == null) throw new WalletClosed()
    if (this._wallet._externalChain.isMyAddress(flawedAddresses[0])) {
        return true;
    }
```

File: src/crypto/wallet.js [hunk 2]
```diff
@@ -1172,6 +1172,36 @@ class WalletManager {
       : CONFIG.CARDANO
     return await api.getPoolInfo(pool, config)
   }
+
+  async checkForFlawedWallets() {
```
Comment (SebastienGllmt) [inline src/crypto/wallet.js]: Remember to add type annotations for function return type

File: src/crypto/wallet.js (hunk 1 above)
Comment (v-almonacid) [reply src/crypto/wallet.js]: Yes we can

**Review by nicarq — APPROVED** [review]

**Pull request merged**